docs/www/guide-wasm-elastic.md

```diff
@@ -0,0 +1,261 @@
+transform.processRecord((recordBatch) => {
+  console.log("Applying Wasm process function")
+  return Promise.all(
+    recordBatch.records.map((record) => {
+      if (containHeader("logService", "elastic", record)) {
+        console.log("Index record value to Elasticsearch")
+        return client.index({
+          index: "result_wasm",
+          type: "redpanda_wasm",
+          body: {
+            /*
+             index the record value into Elasticsearch
+            */
+            record_message: record.value.toString()
+          }
+        })
+      }
+    })
+  ).then(() => {
+      /*
+       return a map with recordBatch indexed to Elasticsearch
+      */
+      const result = new Map();
+      result.set("elasticIndex", recordBatch)
+      /* 
+       Wasm processRecord function has to return a Map always,
+       although that map is empty map
+      */
+      return result
+  })
+  /*
+    Note: if the previous promise failed, the wasm errorHandler is
+    going to handle that error, in this case, Deregister policy will
+    remove this wasm script from wasm engine.    
+  */
+});
+
+exports["default"] = transform;
+```
+
+after applying the new logic, we need to bundle everything into a single file.
+For it, we can use 
+
+```bash
+#this command install all nodejs dependecies
+$ npm install
+#this command generate `dist/elastic.js` file.
+$ npm run build
+```
+
+Before deploying this Wasm script, we need to create the `origin` topic
+The statically defined input topic that this Wasm operator has defined in its metadata
+
+```bash
+rpk topic create origin
+```
+
+ ## Deploy the Wasm function to Redpanda
+
+we use `rpk wasm deploy dist/elastic.js` in this case
+
+```bash
+$ rpk wasm deploy dist/elastic.js
+
+Sent record to partition 0 at offset 1 with timestamp 2021-02-01 18:18:15.734185538 -0500 -05 m=+0.053943881.
+```
+
+To verify that the wasm script has successfully been deployed, 
+you can look through the redpanda logs for a log line that looks like this
+for check it.
+
+on `redpanda logs`
+
+```bash
+INFO [shard 0] coproc - service.cc:101 - Request recieved to register coprocessor with id: 14103244480447969041
+```
+
+The Wasm engine itself also has a log file which is located by default in `/var/lib/redpanda/coprocessor/logs/wasm`. To see the status of your function as reported by the Wasm engine look for log lines like [this]
+
+```bash
+   2021-03-09T14:33:42.367Z [server] info: request enable wasm script:  14103244480447969041
+   2021-03-09T14:33:42.380Z [server] info: wasm script loaded on nodejs engine: 14103244480447969041
+```
+
+## Run data through the Wasm engine
+
+To transform data, push records onto the `origin` topic using any 
+kafka client. In this example we will use the kafka client 
+within `rpk`.
+
+>Note: we need to publish the records with `logService:elastic` 
+>headers in order to those records pass the `containHeader` 
+>validation into `processRecord` function on our wasm script.  
```

Comment:
`in order for those records to pass the 'containHeader' validator that our wasm script will run for each record`